packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -78,39 +79,73 @@ export const styles = theme => {
         top: 'initial',
         bottom: 0,
         width: 2,
+        height: '100%',
       },
     },
     /* Styles applied to the track element before the thumb. */
     trackBefore: {
       zIndex: 1,
       left: 0,
-      transition: trackTransitions,
+      transformOrigin: 'left bottom',
+      '&$rtl': {
+        // left gets rewritten to right which is actually what we want
+        transformOrigin: 'left bottom',
+      },
     },
     /* Styles applied to the track element after the thumb. */
     trackAfter: {
       right: 0,
       opacity: 0.24,
-      transition: trackTransitions,
+      transformOrigin: 'right top',
       '&$vertical': {
-        bottom: 0,
+        top: 0,
+      },
+    },
+    /* Styles applied to the thumb wrapper element. */
+    thumbWrapper: {
+      position: 'relative',
+      zIndex: 2,
+      pointerEvents: 'none',
+      transition: thumbTransitions,
+      '&$activated': {
+        transition: 'none',
+        willChange: 'transform',
+      },
+      '&$rtl': {
+        position: 'relative',
+        left: '100%',
```

Comment:
It took a couple of minutes for me to understand this (especially since it's actually only every applied as `right: 100%`). I think it is simpler to use `translateX(-offset)` as inline style for the `thumbWrapper` when in rtl mode instead of `translateX(+offset)`.

It also has the benfit that we can get rid of `right: unset` in the `$rtl$vertical` rule.